packages/codelabs/src/basics/lit-html.md

```diff
@@ -0,0 +1,1139 @@
+
+To get started, create a simple webpage:
+
+```html
+<!DOCTYPE html>
+<html>
+  <body>
+    My todo app
+  </body>
+</html>
+```
+
+If you run this in the browser and see the message on the screen, you're good to go.
+
+## lit-element setup
+
+lit-element takes care of most of the boilerplate when writing components, providing a great developer experience while staying close to the browser platform. It does not require any build step to run in the browser, and it's only 7kb. This makes it an ideal lightweight choice.
+
+lit-element is written and distributed as es modules, this means we can import it using the browser's native module loader. Let's create a module script, and import LitElement from a CDN:
+
+```html
+<!DOCTYPE html>
+
+<html>
+  <body>
+    <script type="module">
+      import { LitElement } from 'https://unpkg.com/lit-element?module';
+    </script>
+  </body>
+</html>
+```
+
+Make sure you add `type="module"` to the script tag.
+
+Next, we need to define our web component. When writing a vanilla web component we would extend from `HTMLElement`. With lit-element we need to import and extend from `LitElement`, which itself is an extension of `HTMLElement`.
+
+```js
+class TodoApp extends LitElement {
+  connectedCallback() {
+    super.connectedCallback();
+    console.log('lit element connected');
+  }
+}
+
+customElements.define('todo-app', TodoApp);
+```
+
+We defined the tag name for our element as `todo-app`, now we need to add this to the HTML of our page:
+
+```html
+<todo-app></todo-app>
+```
+
+If you run this in the browser you should see `lit element connected` logged to the terminal.
+
+<aside class="notice">
+  Because LitElement also does some work in the connectedCallback, we must always call super.connectedCallback. This is a common source of confusion, so make sure to remember!
+</aside>
+
+Now that we have defined our element, we can start adding a template. lit-element uses lit-html to handle the templating. lit-html works by writing HTML inside of [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) which are strings that can span multiple lines:
+
+```js
+const template = `
+ <h1>Hello world</h1>
+`;
+```
+
+In order to create an actual `lit-html` template, we prefix the template literal with a special HTML tag:
+
+```js
+import { html } from 'https://unpkg.com/lit-element?module';
+
+const template = html`
+  <h1>Hello world</h1>
+`;
+```
+
+This is using a feature called [tagged template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates), where the `html` tag is a function which returns the prepared template ready for rendering. We won't go into details of how it works exactly, but by using this syntax `lit-html` can very efficiently update the dynamic parts of your template when your element re-renders.
```

Comment:
```suggestion
This is a native browser feature called [tagged template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates), where the `html` tag is a function which returns the prepared template ready for rendering. We won't go into details of how it works exactly, but by using this syntax `lit-html` can very efficiently update the dynamic parts of your template when your element re-renders.
```

Reply:
note to self: also update this in the basic web component codelab